### PR TITLE
build(deps): update all non-major dependencies

### LINKED ISSUE #131: build: update all pending library updates

### PR DESCRIPTION
**Pull request by cre8ivejp**
Fixes #131 
- Update AGP version to `8.3.0`
- Update Kotlin version to `1.9.22`

### CODE REVIEW

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/util/intents.kt [hunk 1]
```diff
@@ -2,12 +2,14 @@
...
 package io.bucketeer.sdk.android.internal.util
 
+import android.annotation.SuppressLint
 import android.app.PendingIntent
 import android.content.Context
 import android.content.Intent
 import android.os.Build
 import kotlin.reflect.KClass
 
+@SuppressLint("UnspecifiedImmutableFlag")
```
Comment (duyhungtnn) [inline bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/util/intents.kt:12]: to fix the lint error "Missing PendingIntent mutability flag" which only available from API 23
but still support support API 21

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -48,7 +48,7 @@ android-application = { id = "com.android.application", version.ref = "agp" }
 kotlin-android = { id = "org.jetbrains.kotlin.android", version.ref = "kotlin" }
 kotlin-jvm = { id = "org.jetbrains.kotlin.jvm", version.ref = "kotlin" }
 kotlin-kapt = { id = "org.jetbrains.kotlin.kapt", version.ref = "kotlin" }
-kotlin-dokka = { id = "org.jetbrains.dokka", version = "1.8.20" }
-publish = { id = "com.vanniktech.maven.publish", version = "0.25.3" }
-ksp = { id = "com.google.devtools.ksp", version = "1.8.22-1.0.11" }
+kotlin-dokka = { id = "org.jetbrains.dokka", version = "1.9.20" }
+publish = { id = "com.vanniktech.maven.publish", version = "0.27.0" }
+ksp = { id = "com.google.devtools.ksp", version = "1.9.22-1.0.17" }
 kotlinter = { id = "org.jmailen.kotlinter", version = "3.15.0" }
```
[the remark this answers is not in the document]
Comment (duyhungtnn) [reply gradle/libs.versions.toml:54]: I tried with Kotlinter 4.2.0, but it lead to 100+ errors.
I'd like todo with another PR.

[the remark this answers is not in the document]
Comment (cre8ivejp) [reply gradle/libs.versions.toml:54]: Thank you!

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -1,42 +1,42 @@
 [versions]
-agp = "7.4.2"
+agp = "8.3.0"
 grpc = "1.47.0"
```
[the remark this answers is not in the document]
Comment (duyhungtnn) [reply gradle/libs.versions.toml]: updated in [add: remove unused libs](https://github.com/bucketeer-io/android-client-sdk/pull/135/commits/476ba0ac1bce709219d99d7fd2ddc00a1ee34336)

**Pull request merged**